### ISSUE TITLE
Panic when Consul SD can't be connected to

### ISSUE THREAD

**Issue by brian-brazil** [prometheus/prometheus#1233]
```
time="2015-11-22T17:37:16Z" level=info msg="Loading configuration file /etc/prometheus/prometheus.yml" source="main.go:196" 
panic: discovery.NewConsulDiscovery: Get http://127.0.0.1:8500/v1/agent/self: dial tcp 127.0.0.1:8500: getsockopt: connection refused
```

This shouldn't take out the Prometheus server.

**Comment by wrouesnel** [comment]
+1 just discovered this.

**Comment by fabxc** [comment]
Fixed by #1248
